File: apple/Sources/ReactNativeWatchOS/Modifiers/ListRowBackground.swift
import SwiftUI

/// SwiftUI `.listRowBackground(_:)`. Renders the modifier-content view
/// (hoisted across the bridge under the `content` slot) behind a `List`
/// row. Falls back to the view unchanged when no content was provided.
enum RNWListRowBackgroundModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("listRowBackground") { view, params, ctx in
            guard let body = ctx.content(params.string("content")) else {
                return view
            }
            return AnyView(view.listRowBackground(body))
        }
    }
}

File: src/modifiers/listRowBackground.ts
import type { ReactNode } from 'react';

import { createModifier } from './createModifier';

export interface ListRowBackgroundParams {
  /// The view rendered behind the row's content.
  content: ReactNode;
}

/// SwiftUI `.listRowBackground(_:)`. Places a custom view behind a row in a
/// `List`. The `content` element is hoisted across the bridge by
/// `useModifiers` and rendered as the row background.
export function listRowBackground(params: ListRowBackgroundParams) {
  return createModifier('listRowBackground', params);
}
